test/PlaidClientTest.js

```diff
@@ -594,6 +594,117 @@ describe('plaid.Client', () => {
+          ssn: '999-60-1111',
+          phone_number: '(123)456-7890',
+          email: 'hello@test.com',
+        },
+      };
+      var auditor_id = 'fannie_mae';
+
+      var createAssetReport = (cb) => {
+        pCl.createAssetReport([testAccessToken], days_requested, options,
+        (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.request_id).to.be.ok();
+          expect(response.asset_report_token).to.be.ok();
+          expect(response.asset_report_id).to.be.ok();
+
+          cb(null, response.asset_report_token);
+        });
+      };
+
+      var getAssetReport =
+        (asset_report_token, num_retries_remaining, cb) => {
+        if (num_retries_remaining <= 0) {
+          throw new Error('Ran out of retries while polling for asset report');
+        }
+
+        pCl.getAssetReport(asset_report_token, (err, response) => {
+          if (err) {
+            if (err.status_code === 400 &&
+                err.error_code === 'PRODUCT_NOT_READY') {
+              setTimeout(() => {
+                getAssetReport(
+                  asset_report_token, num_retries_remaining - 1, cb);
+              }, 1000);
+            } else {
+              throw new Error(
+                'Unexpected error while polling for asset report', err);
+            }
+          } else {
+            expect(err).to.be(null);
+            expect(response).to.be.ok();
+            expect(response.report).to.be.ok();
+
+            cb(null, asset_report_token);
+          }
+        });
+      };
+
+      var getAssetReportPdf = (asset_report_token, cb) => {
+        pCl.getAssetReportPdf(asset_report_token, (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+
+          cb(null, asset_report_token);
+        });
+      };
+
+      var createAuditCopy = (asset_report_token, cb) => {
+        pCl.createAuditCopy(asset_report_token, auditor_id,
+          (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.audit_copy_token).to.be.ok();
+
+          cb(null, asset_report_token, response.audit_copy_token);
+        });
+      };
+
+      var removeAuditCopy = (asset_report_token, audit_copy_token, cb) => {
+        pCl.removeAuditCopy(audit_copy_token, (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.removed).to.be(true);
+
+          cb(null, asset_report_token);
+        });
+      };
+
+      var removeAssetReport = (asset_report_token, cb) => {
+        pCl.removeAssetReport(asset_report_token, (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.removed).to.be(true);
+
+          cb();
+        });
+      };
+
+      it('successfully goes through the entire flow', cb => {
+        async.waterfall([
```

Comment:
this test looks good. I don't know if there is any way to provoke the `Warnings` path in sandbox.
I think in terms of asserting API connectivity for each endpoint (which is the principal intention of these tests IMO)... this does the trick 👍

Reply:
Unfortunately not right now :(